src/main/java/com/capgemini/fs/coindashboard/CRUDService/queries/GetQueriesImplementation.java

```diff
@@ -45,6 +45,32 @@ public String getCoinMarketData(
     return gson.toJson(result);
   }
 
+  @Override
+  public String getCoinDetails(String symbol) {
+    MatchOperation matchStage = Aggregation.match(new Criteria("symbol").is(symbol));
+    ProjectionOperation projectStage =
+        Aggregation.project(
+                "name",
+                "symbol",
+                "image_url",
+                "genesis_date",
+                "is_token",
+                "contract_address",
+                "links",
+                "description")
+            .andExclude("_id");
+    Aggregation aggregation =
+        Aggregation.newAggregation(matchStage, projectStage, Aggregation.limit(1));
+    List<Object> result =
+        mongoTemplate.aggregate(aggregation, "Coin", Object.class).getMappedResults();
```

Comment:
I think I have forgotten to point this out in past reviews, but at some point mapping result to `Object` instead of some implementation like `Coin` may be a deficiency. Maybe I'm wrong and Mongo will be sufficient for all oncoming requirements, so here it will be enough to have a thin layer that turns results into JSON (btw. I wonder if Mongo provides some callback mechanism to return JSON without explicit use of GSON).

Reply:
We are using GSON explicite, to make sure that we return only nessesery data in json format. toJson() method provided by MongoTemplate adds Document Structure Descriptions to the data.